src/gui/activityjob.cpp

```diff
@@ -0,0 +1,85 @@
+#include <networkjobs.h>
+#include <qurlquery.h>
+
+#include "activityjob.h"
+
+Q_LOGGING_CATEGORY(lcActivityJob, "nextcloud.gui.activityjob", QtInfoMsg)
+
+namespace OCC {
+
+ActivityJob::ActivityJob(QObject *parent)
+    : QObject(parent)
+{
+}
+
+OcsActivityJob::OcsActivityJob(AccountPtr account, QObject *parent)
+    : ActivityJob(parent)
+    , _account(account)
+{
+}
+
+void OcsActivityJob::queryActivities(const Optional<QString> &objectType, const Optional<QString> &objectId, int limit)
+{
+    _limit = limit;
+    _objectId = objectId;
```

Comment:
can you pass by value and use std::move to avoid doing a copy here
we will always store the parameter

Reply:
I'm not sure I understand the benefit of doing so. Copy will happen in both cases. When using the move there is even an additional move.

Reply:
Thanks. Now I understand.